dashboard/streamlit_graphs/sales_over_time.py

```diff
@@ -107,23 +125,32 @@ def visualize_sales_per_hour(connection: extensions.connection) -> None:
     Visualizes sales per hour within a date range for the Streamlit dashboard.
     """
     default_end = datetime.now()
-    default_start = default_end - timedelta(days=7)
+    default_start = date(2024, 12, 5)
 
     date_range = st.date_input(
-        "Select Date Range:",
-        value=(default_start.date(), default_end.date())
+        "Select a date range:",
+        value=(default_start, default_end),  # Default range
+        min_value=date(2024, 12, 5),  # Earliest selectable date
+        max_value=date.today()       # Latest selectable date
     )
 
     if len(date_range) == 2:
         start_date, end_date = date_range
         if start_date > end_date:
             st.error("Start date must be before or equal to the end date.")
         else:
+            end_date += timedelta(days=1)
```

Comment:
Does adding one day to the end_date always guarantee that it is after the start date?

Reply:
this isn't for making it after the start date; adding one day means that the end date is now included in the graph data range, whereas without it would not.